WallPaperDev/KabegamiTests/CoreDataTests.swift

```diff
@@ -93,13 +96,111 @@ class CoreDataTests: XCTestCase {
       XCTAssertFalse(goal.name == goalName)
     }
   }
+  
+  func testMilestoneCreationToGoal() {
+    let name = "NewGoal"
+    let summary = "NewSummary"
+    let milestoneName = "NewMilestone"
+    let newGoal = manager.createGoal(name, summary)
+    let milestone = manager.createMileStone(milestoneName, newGoal!)
+    let testDate = Date()
+    let milestoneDate = milestone?.createdAt
+    let milestoneDateString = DateFormatter.localizedString(from: milestoneDate!, dateStyle: .short, timeStyle: .none)
+    let testDateString = DateFormatter.localizedString(from: testDate, dateStyle: .short, timeStyle: .none)
+    manager.saveContext()
+    
+    // Check properties of milestone entity
+    XCTAssertNotNil(milestone)
+    XCTAssertNotNil(milestone?.createdAt)
+    XCTAssertNil(milestone?.completedAt)
+    XCTAssertNotNil(milestone?.name)
+    XCTAssertNotNil(milestone?.completed)
+    XCTAssertEqual(milestoneName, milestone?.name!)
+    XCTAssertEqual(false, milestone?.completed)
+    XCTAssertEqual(milestoneDateString, testDateString)
+    XCTAssertNotNil(milestone?.goal)
+    XCTAssertEqual(milestone?.goal, newGoal)
+    
+    // Check milestone related properties in goal entity
+    XCTAssertFalse(newGoal!.milestonesArray.isEmpty)
+    XCTAssertEqual(newGoal?.milestonesArray.count, 1)
+    XCTAssertTrue(newGoal!.milestones!.contains(milestone!))
+  }
+  
+  func testMultipleMilestones() {
+    let name = "NewGoal"
+    let summary = "NewSummary"
+    let newGoal = manager.createGoal(name, summary)!
+    let testMilestones = ["new", "milestone", "names"].compactMap { manager.createMileStone($0, newGoal) }
+    
+    manager.saveContext()
+    XCTAssertFalse(newGoal.milestonesArray.isEmpty)
+    XCTAssertEqual(newGoal.milestonesArray.count, 3)
+    testMilestones.forEach { value in
+      XCTAssertTrue(newGoal.milestones!.contains(value))
+    }
+  }
+  
+  func testUpdateMilestone() {
```

Comment:
🔥 These are great!

Reply:
I just realized now that the tests were giving false positives and I had to rewrite the methods and tests to fix it.